code generators/de.fhdo.lemma.model_processing.code_generation.springcloud.zuul/src/main/java/de/fhdo/lemma/model_processing/code_generation/springcloud/zuul/code_generation/ZuulCodeGenerator.xtend

```diff
@@ -0,0 +1,131 @@
+        val path = '''«targetFolder»«File.separator»«node.name»«File.separator»''' +
+            '''«node.name.toLowerCase»-deployment.yaml'''
+
+        content.put(path, kubernetes)
+    }
+
+    /**
+     * Create an  infrastructure node-specific Docker-Compose deployment file for Zuul.
+     */
+    private def createDockerComposeDeployment(IntermediateInfrastructureNode node) {
+        /*
+         * Set targetFolder for Docker-Compose file and open an existing or create a new
+         * Docker-Compose file
+         */
+        OpenedDockerComposeFile.instance.init(targetFolder)
+        OpenedDockerComposeFile.instance.openExistingDockerComposeFile
+
+        // Build Docker-Compose part based on a operation aspect or node configuration
+        val dockerComposeAspect = node.aspects.findFirst[aspect | aspect.name == "ComposePart"]
+        if (dockerComposeAspect === null)
+            OpenedDockerComposeFile.instance.addOrReplaceDockerComposePart(node.name,
+                ZuulTemplate::getDockerComposeForZuul(node))
+        else
+            OpenedDockerComposeFile.instance.addOrReplaceDockerComposePart(node.name,
+                dockerComposeAspect.propertyValues?.get(0).value)
+
+        // Write Docker-Compose file to the file path
+        val filePath = '''«targetFolder»«File.separator»docker-compose.yml'''
+        content.put(filePath, OpenedDockerComposeFile.instance.toString)
+    }
+
+    /**
+     * Generate a Zuul infrastructure node based on the Spring Cloud framework implemented in
+     * Kotlin.
+     */
+    private def createZuulComponents(IntermediateInfrastructureNode node) {
+        // Generate pom.xml file
+        val pomFilePath = '''«targetFolder»«File.separator»«node.name.toLowerCase»''' +
+            '''«File.separator»pom.xml'''
+        this.content.put(pomFilePath, ZuulTemplate.generatePOMFile(node))
+
+        // Generate main class for the Spring-based Eureka implementation in Kotlin
+        val mainFilePath = '''«targetFolder»«File.separator»«node.name.toLowerCase»''' +
+            buildZuulFilePath(node)
+        this.content.put(mainFilePath, ZuulTemplate::generateZuulApplicationFile(node))
+
+        // Generate Spring-based application.properties file
+        val propertiesFilePath = '''«targetFolder»«File.separator»«node.name.toLowerCase»''' +
+            '''«File.separator»src«File.separator»main«File.separator»resources«File.separator»''' +
+            "application.properties"
+        this.content.put(propertiesFilePath, ZuulTemplate::generateApplicationPropertyFile(node))
+    }
+
+    /**
+     * Build Zuul file path based on the service specific property groupId
+     */
+    private def String buildZuulFilePath(IntermediateInfrastructureNode node) {
+        val property = node.defaultValues.findFirst[
+            it.technologySpecificProperty.name.toLowerCase == "mavengroupid"
+        ]
+        val groupIdPath = property.value.replaceAll("\\.", "/").toLowerCase
```

Comment:
Recall that `replaceAll` works on regular expressions as first argument. Does the call really do what is expected?

Reply:
It transforms the technology specific property `mavenGroupId` from, e.g.,  'de.fhdo.puls.service.zuul' to `de/fhdo/puls/service/zuul`.